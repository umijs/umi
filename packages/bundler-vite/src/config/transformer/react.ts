import reactPlugin from '@vitejs/plugin-react';
import type { IConfigProcessor } from '.';

/**
 * enable react plugin & transform umi babel to vite babel
 */
export default (function react(userConfig) {
  const config: ReturnType<IConfigProcessor> = { plugins: [] };

  // add react plugin config ability.
  if (!userConfig.vue) {
    config.plugins?.push(
      // pre-compiled rollup type is different with installed rollup type
      // so this plugin type is not compatible with config.plugins
      // @ts-ignore
      reactPlugin({
        // jsxRuntime: 'automatic',
        babel: {
          plugins: userConfig.extraBabelPlugins,
          presets: userConfig.extraBabelPresets,
        },
        ...(userConfig?.react || {}),
      }),
    );
  }

  return config;
} as IConfigProcessor);
