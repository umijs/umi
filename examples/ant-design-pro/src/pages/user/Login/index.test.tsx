import { act, render } from '@testing-library/react';
import React from 'react';
import { TestBrowser } from '@@/testBrowser';

const waitTime = (time: number = 100) => {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(true);
    }, time);
  });
};

test('should show login form', async () => {
  const ref = React.createRef<any>();
  let rootContainer = render(
    <TestBrowser
      historyRef={ref}
      location={{
        pathname: '/user/login',
      }}
    />,
  );

  await rootContainer.findAllByText('Ant Design');

  expect(
    rootContainer.baseElement?.querySelector('.ant-pro-form-login-desc')
      ?.textContent,
  ).toBe(
    'Ant Design is the most influential web design specification in Xihu district',
  );

  expect(rootContainer.baseElement).toMatchSnapshot();

  act(() => {
    ref.current?.push('/user/register');
  });

  await rootContainer.findAllByText('注册');

  expect(rootContainer.baseElement?.querySelector('h3')?.textContent).toBe(
    '注册',
  );

  expect(rootContainer.baseElement).toMatchSnapshot();
});
