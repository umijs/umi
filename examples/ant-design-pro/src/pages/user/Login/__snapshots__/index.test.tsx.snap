// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show login form 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="lang"
        data-lang="true"
      >
        <span
          class="ant-dropdown-trigger"
          style="cursor: pointer; padding: 12px; display: inline-flex; align-items: center; justify-content: center; font-size: 18px; vertical-align: middle;"
        >
          <i
            class="anticon"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                class="css-c4d79v"
                d="M12.87 15.07l-2.54-2.51.03-.03c1.74-1.94 2.98-4.17 3.71-6.53H17V4h-7V2H8v2H1v1.99h11.17C11.5 7.92 10.44 9.75 9 11.35 8.07 10.32 7.3 9.19 6.69 8h-2c.73 1.63 1.73 3.17 2.98 4.56l-5.09 5.02L4 19l5-5 3.11 3.11.76-2.04zM18.5 10h-2L12 22h2l1.12-3h4.75L21 22h2l-4.5-12zm-2.62 7l1.62-4.33L19.12 17h-3.24z "
              />
            </svg>
          </i>
        </span>
      </div>
      <div
        class="content"
      >
        <div
          class="ant-pro-form-login-container "
        >
          <div
            class="ant-pro-form-login-top  "
          >
            <div
              class="ant-pro-form-login-header "
            >
              <span
                class="ant-pro-form-login-logo "
              >
                <img
                  alt="logo"
                  src="/logo.svg"
                />
              </span>
              <span
                class="ant-pro-form-login-title "
              >
                Ant Design
              </span>
            </div>
            <div
              class="ant-pro-form-login-desc "
            >
              Ant Design is the most influential web design specification in Xihu district
            </div>
          </div>
          <div
            class="ant-pro-form-login-main "
            style="width: 328px;"
          >
            <div
              class="ant-pro"
            >
              <form
                autocomplete="off"
                class="ant-form ant-form-vertical ant-pro-form"
              >
                <input
                  style="display: none;"
                  type="text"
                />
                <div
                  class="ant-tabs ant-tabs-top"
                >
                  <div
                    class="ant-tabs-nav"
                    role="tablist"
                  >
                    <div
                      class="ant-tabs-nav-wrap"
                    >
                      <div
                        class="ant-tabs-nav-list"
                        style="transform: translate(0px, 0px);"
                      >
                        <div
                          class="ant-tabs-tab ant-tabs-tab-active"
                        >
                          <div
                            aria-selected="true"
                            class="ant-tabs-tab-btn"
                            role="tab"
                            tabindex="0"
                          >
                            Account Login
                          </div>
                        </div>
                        <div
                          class="ant-tabs-tab"
                        >
                          <div
                            aria-selected="false"
                            class="ant-tabs-tab-btn"
                            role="tab"
                            tabindex="0"
                          >
                            Phone Login
                          </div>
                        </div>
                        <div
                          class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                        />
                      </div>
                    </div>
                    <div
                      class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
                    >
                      <button
                        aria-controls="null-more-popup"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-hidden="true"
                        class="ant-tabs-nav-more"
                        id="null-more"
                        style="visibility: hidden; order: 1;"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          aria-label="ellipsis"
                          class="anticon anticon-ellipsis"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="ellipsis"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                            />
                          </svg>
                        </span>
                      </button>
                    </div>
                  </div>
                  <div
                    class="ant-tabs-content-holder"
                  >
                    <div
                      class="ant-tabs-content ant-tabs-content-top"
                    >
                      <div
                        aria-hidden="false"
                        class="ant-tabs-tabpane ant-tabs-tabpane-active"
                        role="tabpanel"
                        tabindex="0"
                      />
                      <div
                        aria-hidden="true"
                        class="ant-tabs-tabpane"
                        role="tabpanel"
                        style="display: none;"
                        tabindex="-1"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                        >
                          <span
                            class="ant-input-prefix"
                          >
                            <span
                              aria-label="user"
                              class="anticon anticon-user prefixIcon"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="user"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                                />
                              </svg>
                            </span>
                          </span>
                          <input
                            class="ant-input ant-input-lg"
                            id="username"
                            placeholder="Username: admin or user"
                            type="text"
                            value=""
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              class="ant-input-clear-icon ant-input-clear-icon-hidden"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="close-circle"
                                class="anticon anticon-close-circle"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="close-circle"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <span
                          class="ant-input-affix-wrapper ant-input-password ant-input-password-large ant-input-affix-wrapper-lg"
                        >
                          <span
                            class="ant-input-prefix"
                          >
                            <span
                              aria-label="lock"
                              class="anticon anticon-lock prefixIcon"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="lock"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                                />
                              </svg>
                            </span>
                          </span>
                          <input
                            action="click"
                            class="ant-input ant-input-lg"
                            id="password"
                            placeholder="Password: ant.design"
                            type="password"
                            value=""
                          />
                          <span
                            class="ant-input-suffix"
                          >
                            <span
                              aria-label="eye-invisible"
                              class="anticon anticon-eye-invisible ant-input-password-icon"
                              role="img"
                              tabindex="-1"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="eye-invisible"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                                />
                                <path
                                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  style="margin-bottom: 24px;"
                >
                  <label
                    class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
                  >
                    <span
                      class="ant-checkbox ant-checkbox-checked"
                    >
                      <input
                        checked=""
                        class="ant-checkbox-input"
                        id="autoLogin"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="ant-checkbox-inner"
                      />
                    </span>
                    <span>
                      Remember me
                    </span>
                  </label>
                  <a
                    style="float: right;"
                  >
                    Forgot Password ?
                  </a>
                </div>
                <button
                  class="ant-btn ant-btn-primary ant-btn-lg"
                  style="width: 100%;"
                  type="button"
                >
                  <span>
                    Login
                  </span>
                </button>
              </form>
            </div>
            <div
              class="ant-pro-form-login-main-other "
            >
              Login with :
              <span
                aria-label="alipay-circle"
                class="anticon anticon-alipay-circle icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="alipay-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M308.6 545.7c-19.8 2-57.1 10.7-77.4 28.6-61 53-24.5 150 99 150 71.8 0 143.5-45.7 199.8-119-80.2-38.9-148.1-66.8-221.4-59.6zm460.5 67c100.1 33.4 154.7 43 166.7 44.8A445.9 445.9 0 00960 512c0-247.4-200.6-448-448-448S64 264.6 64 512s200.6 448 448 448c155.9 0 293.2-79.7 373.5-200.5-75.6-29.8-213.6-85-286.8-120.1-69.9 85.7-160.1 137.8-253.7 137.8-158.4 0-212.1-138.1-137.2-229 16.3-19.8 44.2-38.7 87.3-49.4 67.5-16.5 175 10.3 275.7 43.4 18.1-33.3 33.4-69.9 44.7-108.9H305.1V402h160v-56.2H271.3v-31.3h193.8v-80.1s0-13.5 13.7-13.5H557v93.6h191.7v31.3H557.1V402h156.4c-15 61.1-37.7 117.4-66.2 166.8 47.5 17.1 90.1 33.3 121.8 43.9z"
                  />
                </svg>
              </span>
              <span
                aria-label="taobao-circle"
                class="anticon anticon-taobao-circle icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="taobao-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zM315.7 291.5c27.3 0 49.5 22.1 49.5 49.4s-22.1 49.4-49.5 49.4a49.4 49.4 0 110-98.8zM366.9 578c-13.6 42.3-10.2 26.7-64.4 144.5l-78.5-49s87.7-79.8 105.6-116.2c19.2-38.4-21.1-58.9-21.1-58.9l-60.2-37.5 32.7-50.2c45.4 33.7 48.7 36.6 79.2 67.2 23.8 23.9 20.7 56.8 6.7 100.1zm427.2 55c-15.3 143.8-202.4 90.3-202.4 90.3l10.2-41.1 43.3 9.3c80 5 72.3-64.9 72.3-64.9V423c.6-77.3-72.6-85.4-204.2-38.3l30.6 8.3c-2.5 9-12.5 23.2-25.2 38.6h176v35.6h-99.1v44.5h98.7v35.7h-98.7V622c14.9-4.8 28.6-11.5 40.5-20.5l-8.7-32.5 46.5-14.4 38.8 94.9-57.3 23.9-10.2-37.8c-25.6 19.5-78.8 48-171.8 45.4-99.2 2.6-73.7-112-73.7-112l2.5-1.3H472c-.5 14.7-6.6 38.7 1.7 51.8 6.8 10.8 24.2 12.6 35.3 13.1 1.3.1 2.6.1 3.9.1v-85.3h-101v-35.7h101v-44.5H487c-22.7 24.1-43.5 44.1-43.5 44.1l-30.6-26.7c21.7-22.9 43.3-59.1 56.8-83.2-10.9 4.4-22 9.2-33.6 14.2-11.2 14.3-24.2 29-38.7 43.5.5.8-50-28.4-50-28.4 52.2-44.4 81.4-139.9 81.4-139.9l72.5 20.4s-5.9 14-18.4 35.6c290.3-82.3 307.4 50.5 307.4 50.5s19.1 91.8 3.8 235.7z"
                  />
                </svg>
              </span>
              <span
                aria-label="weibo-circle"
                class="anticon anticon-weibo-circle icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="weibo-circle"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-44.4 672C353.1 736 236 680.4 236 588.9c0-47.8 30.2-103.1 82.3-155.3 69.5-69.6 150.6-101.4 181.1-70.8 13.5 13.5 14.8 36.8 6.1 64.6-4.5 14 13.1 6.3 13.1 6.3 56.2-23.6 105.2-25 123.1.7 9.6 13.7 8.6 32.8-.2 55.1-4.1 10.2 1.3 11.8 9 14.1 31.7 9.8 66.9 33.6 66.9 75.5.2 69.5-99.7 156.9-249.8 156.9zm207.3-290.8a34.9 34.9 0 00-7.2-34.1 34.68 34.68 0 00-33.1-10.7 18.24 18.24 0 01-7.6-35.7c24.1-5.1 50.1 2.3 67.7 21.9 17.7 19.6 22.4 46.3 14.9 69.8a18.13 18.13 0 01-22.9 11.7 18.18 18.18 0 01-11.8-22.9zm106 34.3s0 .1 0 0a21.1 21.1 0 01-26.6 13.7 21.19 21.19 0 01-13.6-26.7c11-34.2 4-73.2-21.7-101.8a104.04 104.04 0 00-98.9-32.1 21.14 21.14 0 01-25.1-16.3 21.07 21.07 0 0116.2-25.1c49.4-10.5 102.8 4.8 139.1 45.1 36.3 40.2 46.1 95.1 30.6 143.2zm-334.5 6.1c-91.4 9-160.7 65.1-154.7 125.2 5.9 60.1 84.8 101.5 176.2 92.5 91.4-9.1 160.7-65.1 154.7-125.3-5.9-60.1-84.8-101.5-176.2-92.4zm80.2 141.7c-18.7 42.3-72.3 64.8-117.8 50.1-43.9-14.2-62.5-57.7-43.3-96.8 18.9-38.4 68-60.1 111.5-48.8 45 11.7 68 54.2 49.6 95.5zm-93-32.2c-14.2-5.9-32.4.2-41.2 13.9-8.8 13.8-4.7 30.2 9.3 36.6 14.3 6.5 33.2.3 42-13.8 8.8-14.3 4.2-30.6-10.1-36.7zm34.9-14.5c-5.4-2.2-12.2.5-15.4 5.8-3.1 5.4-1.4 11.5 4.1 13.8 5.5 2.3 12.6-.3 15.8-5.8 3-5.6 1-11.8-4.5-13.8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="ant-layout-footer"
        style="padding: 0px;"
      >
        <div
          class="ant-pro-global-footer"
        >
          <div
            class="ant-pro-global-footer-list "
          >
            <a
              class="ant-pro-global-footer-list-link "
              href="https://pro.ant.design"
              rel="noreferrer"
              target="_blank"
              title="Ant Design Pro"
            >
              Ant Design Pro
            </a>
            <a
              class="ant-pro-global-footer-list-link "
              href="https://github.com/ant-design/ant-design-pro"
              rel="noreferrer"
              target="_blank"
              title="github"
            >
              <span
                aria-label="github"
                class="anticon anticon-github"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="github"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0138.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
                  />
                </svg>
              </span>
            </a>
            <a
              class="ant-pro-global-footer-list-link "
              href="https://ant.design"
              rel="noreferrer"
              target="_blank"
              title="Ant Design"
            >
              Ant Design
            </a>
          </div>
          <div
            class="ant-pro-global-footer-copyright "
          >
            <span
              aria-label="copyright"
              class="anticon anticon-copyright"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="copyright"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm5.6-532.7c53 0 89 33.8 93 83.4.3 4.2 3.8 7.4 8 7.4h56.7c2.6 0 4.7-2.1 4.7-4.7 0-86.7-68.4-147.4-162.7-147.4C407.4 290 344 364.2 344 486.8v52.3C344 660.8 407.4 734 517.3 734c94 0 162.7-58.8 162.7-141.4 0-2.6-2.1-4.7-4.7-4.7h-56.8c-4.2 0-7.6 3.2-8 7.3-4.2 46.1-40.1 77.8-93 77.8-65.3 0-102.1-47.9-102.1-133.6v-52.6c.1-87 37-135.5 102.2-135.5z"
                />
              </svg>
            </span>
             
            2022 Produced by Ant Financial Experience Department
          </div>
        </div>
      </footer>
    </div>
  </div>
</body>
`;

exports[`should show login form 2`] = `
<body>
  <div>
    <div
      class="main"
    >
      <h3>
        注册
      </h3>
      <form
        class="ant-form ant-form-horizontal"
        id="UserRegister"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="UserRegister_mail"
                  placeholder="邮箱"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="UserRegister_password"
                  placeholder="至少6位密码，区分大小写"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-lg"
                  id="UserRegister_confirm"
                  placeholder="确认密码"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <span
          class="ant-input-group ant-input-group-compact"
        >
          <div
            class="ant-select ant-select-lg ant-select-single ant-select-show-arrow"
            style="width: 20%;"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="+86"
              >
                +86
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
          <div
            class="ant-row ant-form-item"
            style="width: 80%;"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input ant-input-lg"
                    id="UserRegister_mobile"
                    placeholder="手机号"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </span>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-16"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input ant-input-lg"
                      id="UserRegister_captcha"
                      placeholder="验证码"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-col ant-col-8"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-lg getCaptcha"
              type="button"
            >
              <span>
                获取验证码
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <button
                  class="ant-btn ant-btn-primary ant-btn-lg submit"
                  type="submit"
                >
                  <span>
                    注 册
                  </span>
                </button>
                <a
                  class="login"
                  href="/user/login"
                >
                  <span>
                    使用已有账户登录
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;
